### ISSUE TITLE
VirtualMachineImages.getImages() fails when using values from KnownLinuxVirtualMachineImage or KnownWindowsVirtualMachineImage

### ISSUE THREAD

**Issue by davidbairdala** [Azure/azure-sdk-for-java#1392]
Error thrown is 

com.microsoft.azure.CloudException: {
  "error": {
    "code": "InvalidParameter",
    "target": "version",
    "message": "The value of parameter 'version' is invalid."
  }
}

For example, the following groovy code
```
...
def imageRef = KnownLinuxVirtualMachineImage.UBUNTU_SERVER_14_04_LTS.imageReference()

def image = az.virtualMachineImages().getImage(region, imageRef.publisher(), imageRef.offer(), imageRef.sku(), imageRef.version())
...
```

It appears the problem is that the enum contains the string 'latest' for the version, but is not being understood by the getImage api.

Also, would it be possible to add a method 'getImageByImageReference' to VirtualMachineImages?

**Comment by anuchandy** [comment]
@davidbairdala this is a known issue - compute service parse value "latest" for image version and use it as an indication to pick latest image only when we use it in the context of `VM Create` API, using this value on `GetImage` API will not return the latest image.

**Comment by anuchandy** [comment]
@davidbairdala though service does not accept "lastest" as a valid value, found a work around to fetch the latest image. This will be available in the next snapshot/aug release.

**Comment by davidbairdala** [comment]
That's awesome, thanks very much.

**Comment by anuchandy** [comment]
Feature is available as of v1.2 hence closing this.
